runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -7636,7 +7636,7 @@ genInitObjectHeader(TR::Node * node, TR::Instruction *& iCursor, TR_OpaqueClassB
       // TODO: Following approach for initializing object header for array of objects in AOT is conservative.
       // We need support for relocation in generating RIL type instruction. If we have support, we can use
       // same sequence generated in JIT which saves us a load and store.
-      if (comp->compileRelocatableCode())
+      if (comp->compileRelocatableCode() || comp->isOutOfProcessCompilation())
```

Comment:
Changes related to `TR::anewarray` can be removed. I'm able to get passing builds without these checks. So these checks might have been too aggressive.